src/main/java/org/opensearch/security/filter/NettyAttribute.java

```diff
@@ -0,0 +1,39 @@
+package org.opensearch.security.filter;
+
+import java.util.Optional;
+
+import org.opensearch.http.netty4.Netty4HttpChannel;
+import org.opensearch.rest.RestRequest;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.util.AttributeKey;
+
+public class NettyAttribute {
+
+    /**
+     * Gets an attribute value from the request context and clears it from that context
+     */
+    public static <T> Optional<T> popFrom(final RestRequest request, final AttributeKey<T> attribute) {
+        if (request.getHttpChannel() instanceof Netty4HttpChannel) {
+            Channel nettyChannel = ((Netty4HttpChannel) request.getHttpChannel()).getNettyChannel();
```

Comment:
FYI I ran into an issue yesterday where `getHttpChannel` can return a `TraceableHttpChannel`. It does this if the experimental feature flag is enabled:

- https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java#L367
- https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/telemetry/tracing/channels/TraceableHttpChannel.java#L51-L57

Is it possible to create a generic way to get channel attributes regardless of the type of channel that's supplied?

In a previous commit in https://github.com/opensearch-project/OpenSearch/pull/10261 (See [here](https://github.com/cwperks/OpenSearch/blob/aec3ad3dbf5e45c6ee0e79e6752b8b53bd2a72c0/modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpRequestHandler.java#L54-L71)), the `EARLY_RESPONSE` and `CONTEXT_TO_RESTORE` attributes were read in `Netty4HttpRequestHandler` and then fed into the `AbstractHttpServerTransport.incomingRequest` through a `RestHandlerContext` object and used within the `AbstractHttpServerTransport` accordingly. 

Looking through the code, I don't see any test classes that try to run the security plugin with any other http engine other than netty (engine can be set using [http.type](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/network/NetworkModule.java#L93)) so other engines are unknown quantities. If it did need to be engine agnostic, then this could opt to do authn in SecurityRestFilter if the request reaches the REST Filter and the User is not defined in the threadcontext.

Reply:
>Is it possible to create a generic way to get channel attributes regardless of the type of channel that's supplied?

Should be, need to look into this one.